go/private/go_repository.bzl

```diff
@@ -123,7 +123,9 @@ go_repository = repository_rule(
 
 # This is for legacy compatability
 # Originally this was the only rule that triggered BUILD file generation.
-new_go_repository = go_repository
+def new_go_repository(name, **kwargs):
+  print("{0}: new_go_repository has been subsumed into go_repository".format(name))
```

Comment:
Please reword to tell people to migrate. Something like "new_go_repository is deprecated. Please migrate to go_repository soon."